core/entities/Module.php

```diff
@@ -308,7 +308,7 @@ public function saveSetting($setting, $value, $uid = 0, $scope = null)
             return framework\Settings::saveSetting($setting, $value, $this->getName(), $scope, $uid);
         }
 
-        public function deleteSetting($setting, $uid = null, $scope = null)
+        public function deleteSetting($setting, $uid = 0, $scope = null)
```

Comment:
If you change something in here please keep get/save/delete in sync so that their users don't get surprises when they assume they behave in similar matter. With warning logging such thing would need to be added to all three places.

Please consider also that the module API can be used for multiple purposes. So what is correct for this particular case might not be OK for other cases. Eg. providing uid=null for cases where developer wants to get current uid automatically in get/save/delete operations (instead of making query for it outside). Whether such automatic behaviour is good or bad I do not know.